### PR TITLE
OCPBUGSM-46219: OCPBUGSM-46220: Update golang version to 1.18.1.

### PR DESCRIPTION
**Pull request by eliorerz**
Due to several CVE bugs we need to update golang version to >= v1.18.1.
https://bugzilla.redhat.com/show_bug.cgi?id=2077688
https://bugzilla.redhat.com/show_bug.cgi?id=2077689

This PR updates golang version to v1.18.1 and golangci to v1.46.0.

/cc @eliorerz

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -1,6 +1,6 @@
 module github.com/openshift/assisted-installer
 
-go 1.17
+go 1.18
```
Comment (omertuc) [inline go.mod:3]: This will break d/s unless we bump the d/s dockerfile to the 1.18 image

Comment (omertuc) [reply go.mod:3]: Do we already have 1.18 in d/s?

**Review by gamli75 — APPROVED** [review]

**Pull request merged**